### PR TITLE
Use DeckGL CompositeLayer in network map 

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
refactoring

**What is the current behavior?** *(You can also link to an open issue here)*
Network  map layer are not organized and all created in NetworkMap component.

**What is the new behavior (if this is a feature change)?**
DeckGL has composite layer that allows implementing custom layers based on composition of standard layers.
I created 2 business layer: 
  - substation layer: that embed everything regarding substation representation (including substation label)
  - line layer: for lines display

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/components/network/line-layer.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+LineLayer.layerName = 'LineLayer';
+
+LineLayer.defaultProps = {
+    network: null,
+    geoData: null,
+    getNominalVoltageColor: {type: 'accessor', value: [255, 255, 255]},
+    filteredNominalVoltages: [],
+    useName: true
+};
```
Comment (FranckLecuyer) [inline src/components/network/line-layer.js:46]: useName is not used here

File: src/components/network/line-layer.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {CompositeLayer, PathLayer} from 'deck.gl';
+
+class LineLayer extends CompositeLayer {
+
+    renderLayers() {
+        let layers = [];
+
```
Comment (FranckLecuyer) [inline src/components/network/line-layer.js:14]: can be const

File: src/components/network/line-layer.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+LineLayer.layerName = 'LineLayer';
+
+LineLayer.defaultProps = {
+    network: null,
+    geoData: null,
+    getNominalVoltageColor: {type: 'accessor', value: [255, 255, 255]},
+    filteredNominalVoltages: [],
+    useName: true
+};
+
```
Comment (FranckLecuyer) [inline src/components/network/line-layer.js:47]: propTypes is missing

File: src/components/network/substation-layer.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+const SUBSTATION_RADIUS = 500;
+
+function getVoltageLevelRadius(substationRadius, voltageLevel) {
+    return substationRadius / voltageLevel.voltageLevelCount * (voltageLevel.voltageLevelIndex + 1)
+}
+
+class SubstationLayer extends CompositeLayer {
+
+    renderLayers() {
+        let layers = [];
```
Comment (FranckLecuyer) [inline src/components/network/substation-layer.js:19]: can be const

File: src/components/network/substation-layer.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+SubstationLayer.defaultProps = {
+    network: null,
+    geoData: null,
+    getNominalVoltageColor: {type: 'accessor', value: [255, 255, 255]},
+    filteredNominalVoltages: [],
+    useName: true,
+    labelsVisible: false,
+    labelColor: {type: 'color', value: [255, 255, 255]}
+};
+
```
Comment (FranckLecuyer) [inline src/components/network/substation-layer.js:79]: propTypes is missing

File: src/components/network/network-map.js [hunk 6]
```diff
@@ -115,101 +98,48 @@ const NetworkMap = (props) => {
...
                 props.onSubstationClick(info.object.id)
             }
         }
     }
 
-    function getVoltageLevelRadius(substationRadius, voltageLevel) {
-        return substationRadius / voltageLevel.voltageLevelCount * (voltageLevel.voltageLevelIndex + 1)
-    }
-
     let layers = [];
```
Comment (FranckLecuyer) [inline src/components/network/network-map.js:108]: can be const

File: src/components/network/line-layer.js (hunk 1 above)
Comment (geofjamg) [reply src/components/network/line-layer.js:46]: Removed

File: src/components/network/line-layer.js (hunk 2 above)
Comment (geofjamg) [reply src/components/network/line-layer.js:14]: Done

File: src/components/network/line-layer.js (hunk 3 above)
Comment (geofjamg) [reply src/components/network/line-layer.js:47]: There is no propTypes with DeckGL layers. It is similar to React components but not exactly the same and type can be specified only with defaultProps with an optional type field which is in most of the cases infered automatically
https://github.com/uber/deck.gl/blob/master/docs/developer-guide/custom-layers/prop-types.md

File: src/components/network/substation-layer.js (hunk 5 above)
Comment (geofjamg) [reply src/components/network/substation-layer.js:79]: Same

File: src/components/network/substation-layer.js (hunk 4 above)
Comment (geofjamg) [reply src/components/network/substation-layer.js:19]: Done

**Pull request merged**